src/angular-app/languageforge/lexicon/editor/editor.component.ts

```diff
@@ -441,6 +441,21 @@ export class LexiconEditorController implements angular.IController {
     this.goToEntry(id);
   }
 
+  gotoToEntry(index: number, isValid: boolean) {
+    if (isValid) {
+      let id = this.editorService.getIdInFilteredList(Number(index));
+      this.editEntryAndScroll(id);
+      let gotoElement = (<HTMLInputElement>document.getElementById('goto'));
+      gotoElement.value = '';
+    }
+  }
```

Comment:
It feels like the function name should be `goToEntry() instead of `gotoToEntry`  Can you rename?

The if statement around validation isn't very nice, since I believe AngularJS has built-in validation as part of a form using ngSubmit.  e.g. you could wrap the input in a <form ngSubmit="goToEntry()"> and I think the min / max validation will happen for free.  No need to pass in the form validation state.

That's what I'm thinking in my head, however I haven't tried it out.  Let me know how you get on.

Reply:
There already is a function with that name private goToEntry(entryId: string)

Reply:
For the ngSubmit suggestion to work would require an input type of "submit" the text box is input type of "numeric"